src/v2i-hub/PedestrianPlugin/src/PedestrianPlugin.cpp

```diff
@@ -226,37 +247,104 @@ void PedestrianPlugin::OnStateChange(IvpPluginState state)
 	}
 }
 
-void PedestrianPlugin::BroadcastPsm(const std::string &psmJson) 
+const std::string PedestrianPlugin::parseMessage(const std::string& message) {
+    if (message.find("<PersonalSafetyMessage>") != std::string::npos)
+	{
+        return "PersonalSafetyMessage";
+    }
+	else if (message.find("<SensorDataSharingMessage>") != std::string::npos)
+	{
+        return "SensorDataSharingMessage";
+    }
+	else if (message.find("<TravelerInformation>") != std::string::npos)
+	{
+        return "TravelerInformation";
+    }
+	else return "Unknown Message Type";
+}
+
+void PedestrianPlugin::BroadcastPedDet(const std::string &msgJson) 
```

Comment:
Shall we rename this as it now support multiple message types.

Reply:
I renamed everything from PSM to a broader Ped. It's still only generating messages containing Pedestrian information, so I think this is good as is